helix-view/src/editor.rs

```diff
@@ -109,6 +109,69 @@ impl Default for FilePickerConfig {
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum ExplorerStyle {
+    Tree,
+    List,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum ExplorerPosition {
+    Embed,
+    Overlay,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case", default, deny_unknown_fields)]
+pub struct ExplorerConfig {
+    pub style: ExplorerStyle,
+    pub position: ExplorerPosition,
+    /// explorer column width
+    pub column_width: usize,
+}
+
+impl ExplorerConfig {
+    pub fn is_embed(&self) -> bool {
+        match self.position {
```

Comment:
I think all of these util functions can remove `match` since this would mean new variants require updating all the functions to add that match arm. 

E.g.
```rust
pub fn is_embed(&self) -> bool {
    self.position == ExplorerPosition::Embed
}
```